ddsrecorder_participants/src/cpp/recorder/mcap/McapHandler.cpp

```diff
@@ -155,6 +162,11 @@ void McapHandler::add_data(
 {
     std::lock_guard<std::mutex> lock(mtx_);
 
+    DdsTopic topic_ = topic;
```

Comment:
I thought it would be better to just change the scope I know i need to change and not the function which was done with const for a reason. If a concept was initially conceived with const, it's advisable to adhere to that constancy when invoking the function to maintain the integrity of the design.